pkg/agent/proxy/proxier.go

```diff
@@ -422,13 +430,20 @@ func (p *proxier) installServices() {
 			}
 		}
 
-		// If there are expired Endpoints, Endpoints installed should be updated.
-		if internalNodeLocal && externalNodeLocal && len(localEndpointUpdateList) < len(endpointsInstalled) ||
-			!(internalNodeLocal && externalNodeLocal) && len(allEndpointUpdateList) < len(endpointsInstalled) {
+		// If there are expired Endpoints, installed flows for Endpoints should be updated.
+		if internalNodeLocal && (externalNodeLocal || svcInfo.NodePort() == 0) && len(localEndpointUpdateList) < len(endpointsInstalled) ||
+			(!internalNodeLocal || !externalNodeLocal && svcInfo.NodePort() > 0) && len(allEndpointUpdateList) < len(endpointsInstalled) {
```

Comment:
In what case this condition `!externalNodeLocal && svcInfo.NodePort() > 0` will be true?

Reply:
This part was modified after rebasing to the latest code.

For this question, when a NodePort or LoadBalancer,  when its externalTrafficPolicy is Cluster `!externalNodeLocal && svcInfo.NodePort() > 0` will be true.